src/audio/multiband_drc/multiband_drc.c

```diff
@@ -126,6 +186,67 @@ static int multiband_drc_init_coef(struct processing_module *mod, int16_t nch, u
 	comp_info(dev, "multiband_drc_init_coef(), initializing %i-way crossover",
 		  config->num_bands);
 
+	/* Calculate total size for the contiguous memory block */
+	size_t total_size = 0;
+
+	for (ch = 0; ch < nch; ++ch) {
+		size_t emp_deemp_size = sizeof(struct sof_eq_iir_biquad) * SOF_EMP_DEEMP_BIQUADS;
+		size_t crossover_delay_size = sizeof(uint64_t) * CROSSOVER_NUM_DELAYS_LR4;
+
+		/*  For first block (emphasis)*/
+		total_size += emp_deemp_size + crossover_delay_size;
+		/*  For second block (crossover delays)*/
+		total_size += emp_deemp_size + crossover_delay_size;
+		/*  For third block (deemphasis)*/
+		total_size += emp_deemp_size + crossover_delay_size;
+
+		/* Each block needs to be properly aligned */
+		total_size += (emp_deemp_size % __alignof__(struct sof_eq_iir_biquad) != 0) ?
+			__alignof__(struct sof_eq_iir_biquad) -
+			(emp_deemp_size % __alignof__(struct sof_eq_iir_biquad)) : 0;
+
+		total_size += (crossover_delay_size % __alignof__(uint64_t) != 0) ?
+			__alignof__(uint64_t) -
+			(crossover_delay_size % __alignof__(uint64_t)) : 0;
+	}
+
+	/* For DRC states */
+	total_size += sizeof(struct drc_state) * num_bands * nch;
+
+	/* Define offsets and pointer lists for allocations */
+	void *ptrs[3 * nch * 3 + num_bands];
+	size_t sizes[3 * nch * 3 + num_bands];
+
+	for (ch = 0; ch < nch; ++ch) {
```

Comment:
rather than doing this convoluted offset loop, why not simply store your offsets for each field, then do base_addr + offset and store it directly to the field in the struct.